plugin.lua

```diff
@@ -1,185 +1,151 @@
-local function not_has(table, val)
-    for _, value in pairs(table) do
-        if value == val then
-            return false
-        end
-    end
-
-    return true
-end
+ITEM_WIDTH = 200
+WINDOW_WIDTH = 275
 
 function draw()
-    imgui.Begin("Unsnapped Notes Finder")
 
+    imgui.SetNextWindowSizeConstraints({WINDOW_WIDTH, 0}, {WINDOW_WIDTH, 4000})
+    imgui.Begin("Unsnapped Notes Finder")
     state.IsWindowHovered = imgui.IsWindowHovered()
 
-    local n1 = state.GetValue("n1") or 12
-    local n2 = state.GetValue("n2") or 16
-    local leniency = state.GetValue("leniency") or 0
-
-    if n1 < 1 then n1 = 1 end
-    if n2 < 1 then n2 = 1 end
-    if leniency < 0 then leniency = 0 end
-
-    local errorstring = state.GetValue("errorstring") or ""
-
-    local errors = state.GetValue("errors") or {}
-
-    local currentobject = state.GetValue("currentobject") or 0
-
-    local showcopytext = state.GetValue("showcopytext") or false
-
-    --local debug = state.GetValue("debug") or "hi"
+    local vars = {
+        snaps = "16, 12",
+        leniency = 1,
+        deltaInfo = {},
+        selectedOnly = false
+    }
 
-    _, n1 = imgui.InputInt("1/n Snap", n1)
-    _, n2 = imgui.InputInt("1/n Snap ", n2)
+    getVars(vars)
 
-    _, leniency = imgui.InputInt("Lenience", leniency)
+    imgui.PushItemWidth(ITEM_WIDTH)
 
-    if imgui.Button("Search") then
-        local tps = map.TimingPoints
-        local notes = map.HitObjects
+    _, vars.snaps = imgui.InputText("Snaps", vars.snaps, 50)
+    vars.snaps = restrictSnapString(vars.snaps)
 
-        errors = {}
+    _, vars.leniency = imgui.InputInt("Leniency", vars.leniency)
+    vars.leniency = math.max(vars.leniency, 0) -- no negative numbers
```

Comment:
make min leniency 1 and not 0 to prevent idiots like myself from lagging out the editor
```suggestion
    vars.leniency = math.max(vars.leniency, 1) -- no negative numbers
```